Gems/AtomLyIntegration/AtomRenderOptions/Code/Source/AtomRenderOptions.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "AtomRenderOptions.h"
+
+#include <AzCore/Name/Name.h>
+
+#include <Atom/RPI.Public/Base.h>
+#include <Atom/RPI.Public/Pass/PassFilter.h>
+#include <Atom/RPI.Public/RPISystemInterface.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+
+namespace AZ::Render
+{
+    RPI::RenderPipelinePtr GetDefaultViewportPipelinePtr()
+    {
+        RPI::RPISystemInterface* rpiSystem = RPI::RPISystemInterface::Get();
+        if (!rpiSystem)
+        {
+            return nullptr;
+        }
+
+        AzFramework::NativeWindowHandle windowHandle = RPI::ViewportContextRequests::Get()->GetDefaultViewportContext()->GetWindowHandle();
+        if (!windowHandle)
+        {
+            return nullptr;
+        }
+
+        return rpiSystem->GetRenderPipelineForWindow(windowHandle);
+    }
+
+    AZStd::optional<bool> IsPassEnabled(const Name& name)
+    {
+        RPI::RenderPipelinePtr pipeline = GetDefaultViewportPipelinePtr();
+        if (!pipeline)
+        {
+            return AZStd::nullopt;
+        }
+
+        RPI::Pass* pass = RPI::PassSystemInterface::Get()->FindFirstPass(RPI::PassFilter::CreateWithPassName(name, pipeline.get()));
+        if (!pass)
+        {
+            return AZStd::nullopt;
+        }
+
+        return pass->IsEnabled();
+    }
```

Comment:
Couldn't this just be bool and return false if there is no pipeline or the pass is not found? Maybe there should be a second function to check if a pass exists or is valid.

Reply:
Done :)